// coloredtext/colors.js
const def = "\\x1b[";
exports.raw = {
	alloff: `${def}0m`,
	bold: `${def}1m`,
	underline: `${def}4m`,
	blink: `${def}5m`,
	boldoff: `${def}21m`,
	undelineoff: `${def}24m`,
	blinkoff: `${def}25m`,
	all: [
		"alloff",
		"bold",
		"underline",
		"blink",
		"boldoff",
		"underlineoff",
		"blinkoff",
		"all",
		"list"
	],
	list: [
		"alloff",
		"bold",
		"underline",
		"blink",
		"boldoff",
		"underlineoff",
		"blinkoff",
		"all",
		"list"
	],
	bg: {
		black: `${def}40m`,
		red: `${def}41m`,
		green: `${def}42m`,
		yellow: `${def}43m`,
		blue: `${def}44m`,
		magenta: `${def}45m`,
		cyan: `${def}46m`,
		white: `${def}47m`,
		default: `${def}49m`,
		gray: `${def}100m`,
		grey: `${def}100m`,
		lightred: `${def}101m`,
		lightgreen: `${def}102m`,
		lightyellow: `${def}103m`,
		lightblue: `${def}104m`,
		lightmagenta: `${def}105m`,
		lightcyan: `${def}106m`,
		lightwhite: `${def}107m`,
		all: [
			"black",
			"red",
			"green",
			"yellow",
			"blue",
			"magenta",
			"cyan",
			"white",
			"default",
			"gray",
			"grey",
			"lightgray",
			"lightgrey",
			"lightred",
			"lightgreen",
			"lightyellow",
			"lightblue",
			"lightmagenta",
			"lightcyan",
			"lightwhite",
			"all",
			"list"
		],
		list: [
			"black",
			"red",
			"green",
			"yellow",
			"blue",
			"magenta",
			"cyan",
			"white",
			"default",
			"gray",
			"grey",
			"lightgray",
			"lightgrey",
			"lightred",
			"lightgreen",
			"lightyellow",
			"lightblue",
			"lightmagenta",
			"lightcyan",
			"lightwhite",
			"all",
			"list"
		],
	},
	fg: {
		black: `${def}30m`,
		red: `${def}31m`,
		green: `${def}32m`,
		yellow: `${def}33m`,
		blue: `${def}34m`,
		magenta: `${def}35m`,
		cyan: `${def}36m`,
		white: `${def}37m`,
		default: `${def}39m`,
		gray: `${def}90m`,
		grey: `${def}90m`,
		lightred: `${def}91m`,
		lightgreen: `${def}92m`,
		lightyellow: `${def}93m`,
		lightblue: `${def}94m`,
		lightmagenta: `${def}95m`,
		lightcyan: `${def}96m`,
		lightwhite: `${def}97m`,
		all: [
			"black",
			"red",
			"green",
			"yellow",
			"blue",
			"magenta",
			"cyan",
			"white",
			"default",
			"gray",
			"grey",
			"lightgray",
			"lightgrey",
			"lightred",
			"lightgreen",
			"lightyellow",
			"lightblue",
			"lightmagenta",
			"lightcyan",
			"lightwhite",
			"all",
			"list"
		],
		list: [
			"black",
			"red",
			"green",
			"yellow",
			"blue",
			"magenta",
			"cyan",
			"white",
			"default",
			"gray",
			"grey",
			"lightgray",
			"lightgrey",
			"lightred",
			"lightgreen",
			"lightyellow",
			"lightblue",
			"lightmagenta",
			"lightcyan",
			"lightwhite",
			"all",
			"list"
		],
	},
}

exports.alloff = txt => {
	if (txt == undefined) return undefined;
	return "\x1b[0m" + txt;
}
exports.bg = {
	black: txt => {
		if (txt == undefined) return undefined;
		return "\x1b[40m" + txt + "\x1b[0m";
	},
	blue: txt => {
		if (txt == undefined) return undefined;
		return "\x1b[44m" + txt + "\x1b[0m";
	},
	cyan: txt => {
		if (txt == undefined) return undefined;
		return "\x1b[46m" + txt + "\x1b[0m";
	},
	gray: txt => {
		if (txt == undefined) return undefined;
		return "\x1b[100m" + txt + "\x1b[0m";
	},
	green: txt => {
		if (txt == undefined) return undefined;
		return "\x1b[42m" + txt + "\x1b[0m";
	},
	grey: txt => {
		if (txt == undefined) return undefined;
		return "\x1b[100m" + txt + "\x1b[0m";
	},
	lightblue: txt => {
		if (txt == undefined) return undefined;
		return "\x1b[104m" + txt + "\x1b[0m";
	},
	lightcyan: txt => {
		if (txt == undefined) return undefined;
		return "\x1b[106m" + txt + "\x1b[0m";
	},
	lightgreen: txt => {
		if (txt == undefined) return undefined;
		return "\x1b[102m" + txt + "\x1b[0m";
	},
	lightred: txt => {
		if (txt == undefined) return undefined;
		return "\x1b[101m" + txt + "\x1b[0m";
	},
	lightyellow: txt => {
		if (txt == undefined) return undefined;
		return "\x1b[103m" + txt + "\x1b[0m";
	},
	lightmagenta: txt => {
		if (txt == undefined) return undefined;
		return "\x1b[105m" + txt + "\x1b[0m";
	},
	lightwhite: txt => {
		if (txt == undefined) return undefined;
		return "\x1b[107m" + txt + "\x1b[0m";
	},
	magenta: txt => {
		if (txt == undefined) return undefined;
		return "\x1b[45m" + txt + "\x1b[0m";
	},
	red: txt => {
		if (txt == undefined) return undefined;
		return "\x1b[41m" + txt + "\x1b[0m";
	},
	yellow: txt => {
		if (txt == undefined) return undefined;
		return "\x1b[43m" + txt + "\x1b[0m";
	},
	white: txt => {
		if (txt == undefined) return undefined;
		return "\x1b[47m" + txt + "\x1b[0m";
	}
}
exports.black = txt => {
	if (txt == undefined) return undefined;
	return "\x1b[30m" + txt + "\x1b[0m";
}
exports.blink = txt => {
	if (txt == undefined) return undefined;
	return "\x1b[5m" + txt;
}
exports.blinkoff = txt => {
	if (txt == undefined) return undefined;
	return "\x1b[25m" + txt;
}
exports.blue = txt => {
	if (txt == undefined) return undefined;
	return "\x1b[34m" + txt + "\x1b[0m";
}
exports.bold = txt => {
	if (txt == undefined) return undefined;
	return "\x1b[1m" + txt + "\x1b[0m";
}
exports.boldoff = txt => {
	if (txt == undefined) return undefined;
	return "\x1b[21m" + txt + "\x1b[0m";
}
exports.cyan = txt => {
	if (txt == undefined) return undefined;
	return "\x1b[36m" + txt + "\x1b[0m";
}
exports.green = txt => {
	if (txt == undefined) return undefined;
	return "\x1b[32m" + txt + "\x1b[0m";
}
exports.lightblue = txt => {
	if (txt == undefined) return undefined;
	return "\x1b[94m" + txt + "\x1b[0m";
}
exports.lightcyan = txt => {
	if (txt == undefined) return undefined;
	return "\x1b[96m" + txt + "\x1b[0m";
}
exports.gray = txt => {
	if (txt == undefined) return undefined;
	return "\x1b[90m" + txt + "\x1b[0m";
}
exports.lightgreen = txt => {
	if (txt == undefined) return undefined;
	return "\x1b[92m" + txt + "\x1b[0m";
}
exports.grey = txt => this.gray(txt);
exports.lightred = txt => {
	if (txt == undefined) return undefined;
	return "\x1b[91m" + txt + "\x1b[0m";
}
exports.lightyellow = txt => {
	if (txt == undefined) return undefined;
	return "\x1b[93m" + txt + "\x1b[0m";
}
exports.lightmagenta = txt => {
	if (txt == undefined) return undefined;
	return "\x1b[95m" + txt + "\x1b[0m";
}
exports.lightwhite = txt => {
	if (txt == undefined) return undefined;
	return "\x1b[97m" + txt + "\x1b[0m";
}
exports.magenta = txt => {
	if (txt == undefined) return undefined;
	return "\x1b[35m" + txt + "\x1b[0m";
}
exports.red = txt => {
	if (txt == undefined) return undefined;
	return "\x1b[31m" + txt + "\x1b[0m";
}
exports.reset = txt => {
	if (txt == undefined) return undefined;
	return "\x1b[39m" + txt + "\x1b[0m";
}
exports.underline = txt => {
	if (txt === undefined) return undefined;
	return "\x1b[4m" + txt + "\x1b[0m";
}
exports.underlineoff = txt => {
	if (txt == undefined) return undefined;
	return "\x1b[24m" + txt + "\x1b[0m";
}
exports.yellow = txt => {
	if (txt == undefined) return undefined;
	return "\x1b[33m" + txt + "\x1b[0m";
}
exports.white = txt => {
	if (txt == undefined) return undefined;
	return "\x1b[37m" + txt + "\x1b[0m";
}
